### PR TITLE
Add alicekk support

### PR DESCRIPTION
**Pull request by TW59420**
## Description
MWStudio is a custom keyboard enthusiast
Upload qmk firmware of alicekk series PCB

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mwstudio/alicekk/rules.mk [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no      # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/mwstudio/alicekk/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/mwstudio/alicekk/rules.mk [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# RGB_MATRIX_DRIVER = WS2812
+# WS2812_DRIVER = bitbang
+#WS2812_DRIVER = pwm
+
+# STM32_EEPROM_ENABLE = yes
+# RGB_MATRIX_ENABLE = yes
+# RGB_MATRIX_DRIVER = WS2812
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+# OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (drashna) [inline keyboards/mwstudio/alicekk/rules.mk:34]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/alicekk/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# AliceKK
+
+![AliceKK](https://imgur.com/a/LPzk032)
```
Comment (fauxpark) [inline keyboards/mwstudio/alicekk/readme.md:3]:
```suggestion
![AliceKK](https://i.imgur.com/xfjynwgh.jpeg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/alicekk/config.h [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#define USB_POLLING_INTERVAL_MS  2
+#define QMK_KEYS_PER_SCAN  12
+
+#define RGB_DI_PIN B1
+
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200
+#define RGB_MATRIX_STARTUP_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS
+#define RGBLED_NUM 87
+
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/mwstudio/alicekk/config.h:57]:
```suggestion
```

File: keyboards/mwstudio/alicekk/alicekk.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "alicekk.h"
```
Comment (drashna) [inline keyboards/mwstudio/alicekk/alicekk.c:16]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "alicekk.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

File: keyboards/mwstudio/alicekk/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    action_exec(encoder_event);
+}
+
+void matrix_scan_user(void) {
+    encoder_action_unregister();
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    encoder_action_register(index, clockwise);
+    return true;
```
Comment (drashna) [inline keyboards/mwstudio/alicekk/keymaps/via/keymap.c:94]:
```suggestion
    return false;
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/alicekk/config.h [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+#define DIODE_DIRECTION ROW2COL
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+#define FORCE_NKRO
+
+/* 将USB 轮询速率更改为 1000hz 并为精英游戏每次扫描使用更大的密钥*/
+#define USB_POLLING_INTERVAL_MS  2
+#define QMK_KEYS_PER_SCAN  12
```
Comment (drashna) [inline keyboards/mwstudio/alicekk/config.h:49]: polling rate is set to 1 by default, actually.  And the keys per scan probably shouldn't be used
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/mwstudio/alicekk/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                 _______, _______,          _______, _______,                   _______,          _______,          _______, _______, _______
+    ),
+
+	[3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+                 _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                 _______, _______,          _______, _______,                   _______,          _______,          _______, _______, _______
+    ),
```
Comment (zvecr) [inline keyboards/mwstudio/alicekk/keymaps/default/keymap.c:52]:
```suggestion
```

**Review by zvecr — APPROVED** [review]

**Pull request merged**